src/wallet/password_container.cpp

```diff
@@ -297,5 +159,90 @@ bool password_container::read_from_tty_double_check(const char *message) {
     return true;
   }
 
-#endif
-}
+#endif // end !WIN32
+
+  void clear(std::string& pass) noexcept
+  {
+    //! TODO Call a memory wipe function that hopefully is not optimized out
```

Comment:
`memset_s` is not available on my glibc (2.22) or Ubuntu 16.04. It is an optional part of the C11 standard, and AFAIK is not listed in the C++11 or C++14 standard. I _think_ the only method is attempting to call various platform specific functions that claim to never be optimized away. libsodium has a function which does exactly this, with two fallback strategies.

Apparently `memset_s` exists somewhere if libsodium has detection for it - anybody know the target? Also, a worthwhile read on whether [wiping memory is currently futile in C](http://www.daemonology.net/blog/2014-09-06-zeroing-buffers-is-insufficient.html). Although, I see no harm in making a "best effort" to wipe sensitive memory.